ash/src/entry_libloading.rs

```diff
@@ -52,7 +52,11 @@ impl From<MissingEntryPoint> for LoadingError {
     }
 }
 
-/// Default function loader
+/// Runtime function loader
+///
+/// Prefer [`EntryLinked`](crate::EntryLinked) in code that would would otherwise panic on
+/// [`Entry::new`] failing.
```

Comment:
This is my biggest concern as the Vulkan library will need to be present during linking and thus break CIs. One could indeed install the Vulkan Loader on the CI server somehow but i think that's too much of a hassle.

Reply:
@Friz64 I'm curious if this isn't the case already. GH Actions - which a lot of Rust crates use - is already loaded to the brim with packages.

Reply:
Good thought, but [does not look like that is the case, unfortunately](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md).

Reply:
> If all these questions are answered with no we should probably seek to move in this direction by default, instead of using libloading by default.

I agree. The only reason I'm aware of to use dlopen for Vulkan is if you have a graceful fallback path, which is an extremely uncommon case, well suited to requiring an opt-in.

> One could indeed install the Vulkan Loader on the CI server somehow but i think that's too much of a hassle.

If this is an issue, fixing it is as simple as `apt-get install libvulkan1` or whatever--trivial. But CI seems to be passing fine with `--all-features` already. The loader is part of the base install of most distros these days, after all.

Reply:
> I agree. The only reason I'm aware of to use dlopen for Vulkan is if you have a graceful fallback path, which is an extremely uncommon case, well suited to requiring an opt-in.

Exactly this :+1: 
 
> > One could indeed install the Vulkan Loader on the CI server somehow but i think that's too much of a hassle.
> 
> If this is an issue, fixing it is as simple as `apt-get install libvulkan1` or whatever--trivial. But CI seems to be passing fine with `--all-features` already. The loader is part of the base install of most distros these days, after all.

I couldn't find it in the same page @Friz64 linked, and `--all-features` on `check` does not perform the final linking step where the library is needed, we should use `cargo build` for that. That's probably a good thing, and for projects needing to spit out built artifacts they can easily add it with `apt` - that's the justification for [GH actions not including `Vulkan` (the SDK) in the first place](https://github.com/actions/virtual-environments/issues/18).

Reply:
Switched to linking by default.